pandablocks/utils.py

```diff
@@ -60,11 +60,18 @@ def words_to_table(
         if field_info.subtype == "int":
             # First convert from 2's complement to offset, then add in offset.
             packing_value = (value ^ (1 << (bit_length - 1))) + (-1 << (bit_length - 1))
+            packing_value = packing_value.astype(np.int32)
         elif field_info.subtype == "enum" and convert_enum_indices:
             assert field_info.labels, f"Enum field {field_name} has no labels"
             packing_value = [field_info.labels[x] for x in value]
         else:
-            packing_value = value
+            # Use shorter types, as these are used in waveform creation
+            if bit_length <= 8:
+                packing_value = value.astype(np.uint8)
+            elif bit_length <= 16:
+                packing_value = value.astype(np.uint16)
+            else:
+                packing_value = value.astype(np.uint32)  # already uint32
```

Comment:
What was the conclusion to the discussion we had about converting these to `int8/int16` types? That would mean the records that are created from these types would be given `CHAR` or `SHORT` types, and so Channel Access would no longer have to convert to `DOUBLE` to be able to transmit these values

Reply:
The uint32 are converted to double still. But yea the uint8/uint16 are char/short types.